### ISSUE TITLE
Per province/state time series graphs with series synchronization

### ISSUE THREAD

**Issue by afrantzis** [CSSEGISandData/COVID-19#346]
Hi everyone and thank you for all the great work!

I wanted a visual and interactive way to see how trends in various countries compared to each other, so I created a light and simple graph web page at:

 https://afrantzis.com/covid-19/

You can plot the time series per state/province normally, but, in addition, you can "synchronize" the time series lines so that they start approximately at a (user) specified case count. This effectively overlays the time series by removing the time offset between them, which makes eyeballing trends easier. Hope others find this useful, too!

**Comment by 2fst4u** [comment]
It would be good if all the seperated provinces were grouped into countries.

**Comment by rtwfroody** [comment]
I built something very similar: http://casualhacker.net/covid19/

I haven't done synchronization, but I have been thinking I want to.

**Comment by afrantzis** [comment]
@2fst4u Yeah, the province/state separation is quite inconsistent at the moment, so I have changed the graph to group by whole countries instead. Thanks!

**Comment by afrantzis** [comment]
@rtwfroody Nice! I like the idea of being able to display the data relative to the population; it provides a different and interesting perspective on the trends.

**Comment by Swartzlander** [comment]
I've been posting semi-log data to distinguish exponential growth curves from other dynamic behavior such as bubble phases. Representative doubling times are also provided.
The URL is
http://www.cis.rit.edu/~grovers/COVID/Coronavirus_to_01Mar20.jpg

This plot will be updated daily as time allows.  The file name will not change.

**Comment by nrhine1** [comment]
I've created some Python utilities to enable this type of plotting.

https://github.com/nrhine1/COVID-19-pyvis

**Comment by Bost** [comment]
> I built something very similar: http://casualhacker.net/covid19/

> https://afrantzis.com/covid-19/

@rtwfroody @afrantzis :+1: Would you share your code please? Thanks.

**Comment by rtwfroody** [comment]
@Bost, my code is at https://github.com/rtwfroody/c19growth

**Comment by afrantzis** [comment]
Hi everyone, I have made a few updates to the web page, which I hope people will find useful/interesting:

1. Added an automatic synchronization option to try to get the best trend match for all displayed time series.
2. Show how much each series has shifted during synchronization (e.g., "France (shifted left 6 days)").
3. Manual synchronization (i.e., sync at a specific case count) now maintains the whole range on the x-axis (i.e., no longer changes the range to start from the sync point).

**Comment by afrantzis** [comment]
@Bost Hi! My code is at https://github.com/afrantzis/covid-19/.
